pkg/controller/controller_utils.go

```diff
@@ -602,7 +602,12 @@ func (r RealPodControl) DeletePod(namespace string, podID string, object runtime
 		return fmt.Errorf("object does not have ObjectMeta, %v", err)
 	}
 	klog.V(2).Infof("Controller %v deleting pod %v/%v", accessor.GetName(), namespace, podID)
-	if err := r.KubeClient.CoreV1().Pods(namespace).Delete(context.TODO(), podID, metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
+	if err := r.KubeClient.CoreV1().Pods(namespace).Delete(context.TODO(), podID, metav1.DeleteOptions{}); err != nil {
```

Comment:
The not found initially comes from https://github.com/kubernetes/kubernetes/pull/67826 to prevent repeating failure events based on stale caches. We shouldn't swallow the error, but we should avoid the event I think.